core/tree/utils.go

```diff
@@ -0,0 +1,233 @@
+package tree
```

Comment:
instead of the name utils.go I suggest to rename this to print.go (carving out the optimization sort, ofc)

Reply:
moved